### PR TITLE
Update TMS & CMS HandShake and AES Support Poxy(socks5) and Fix Import MSniffer logfile error

### PR DESCRIPTION
**Pull request by pungin**

### CODE REVIEW

File: StructureSegment.cs [hunk 1]
```diff
@@ -5,12 +5,35 @@
...
+                case MapleLocale.KOREA_TEST:
+                    ascii = Encoding.GetEncoding("EUC_KR");
+                    break;
+                case MapleLocale.JAPAN:
+                    ascii = Encoding.GetEncoding("Shift_JIS");
+                    break;
+                case MapleLocale.CHINA:
+                    ascii = Encoding.GetEncoding("GB18030");
+                    break;
+                case MapleLocale.TESPIA:
```
Comment (diamondo25) [inline StructureSegment.cs]: cool!

File: SessionForm.cs [hunk 2]
```diff
@@ -120,28 +123,94 @@ internal Results BufferTCPPacket(TcpPacket pTCPPacket, DateTime pArrivalTime)
...
+                        }
+                        socks5++;
+                        return Results.Continue;
+                    }
+                    else if (tcpData.Length == 3 && pr.ReadByte() == 5)
+                    {
+                        socks5 = 1;
+                        return Results.Continue;
+                    }
+                    Console.WriteLine(mLocalEndpoint + "Not a MapleStory HankShake");
```
Comment (diamondo25) [inline SessionForm.cs]: HankShake :)? Also, please use formatting:
```
Console.WriteLine("Connection on port {0} did not have a MapleStory Handshake", mLocalEndpoint);
```

File: MainForm.cs [hunk 3]
```diff
@@ -491,8 +492,8 @@ private void setupToolStripMenuItem_Click(object sender, EventArgs e)
         private void importMSnifferToolStripMenuItem_Click(object sender, EventArgs e)
         {
             OpenFileDialog ofd = new OpenFileDialog();
-            ofd.Title = "Select MSniffer logfile";
-            ofd.Filter = "*.*";
+            ofd.Title = "Select MSniffer logfile";
+            ofd.Filter = "AllFile|*.*";
```
Comment (diamondo25) [inline MainForm.cs]: Please change this to `"All files|*.*"`

**Comment by diamondo25** [comment]
Hey @pungin, thank you for your additions to MapleShark!
I've added some comments that I would like to hear from you about.
After that's cleared up, I can merge it into master :)

**Comment by pungin** [comment]
I agree with you @diamondo25 .
My English is not well, so please understanding.

**Comment by blackmaple2018** [comment]
@diamondo25 hey , can u help me update this tool to support opcode encryption in new maplestory version, thanks ddd,  i'm chinese , sorry for my bad english,,

**Pull request merged**